docs/en/api/FS-API.md

```diff
@@ -78,6 +78,36 @@ CreateFilePOptions options = CreateFilePOptions.newBuilder().setBlockSizeBytes(6
 FileOutStream out = fs.createFile(path, options);
 ```
 
+#### Programmatically Modifying Configuration
+
+Alluxio configuration can be set through alluxio-site.properties, but these properties apply to all
+instances of Alluxio that read from the file. If fine-grained configuration management is required,
+pass in a customized configuration object when creating the `FileSystem` object.
+The generated `FileSystem` object will have modified configuration properties, independent of any
+other `FileSystem` clients.
+
+```java
+FileSystem normalFs = FileSystem.Factory.get();
+AlluxioURI normalPath = new AlluxioURI("/normalFile");
+// Create a file with default properties
+FileOutStream normalOut = normalFs.createFile(normalPath);
+...
+out.close();
+
+// Create a file system with custom configuration
+InstancedConfiguration conf = new InstancedConfiguration(ConfigurationUtils.defaults());
```

Comment:
- It is really uncommon to see `XXXXUtils.defaults`. Why not to provide a factory method `InstancedConfiguration.defaults()` directly? --- I feel really strong about it from a developer's perspective.
- Since original `Configuration` interface is no more used, can we rename `InstancedConfiguration` to `Configuration` to make it easy to write?

Reply:
These require code changes which would take a fair amount of effort to overhaul. For being very close to the release this is probably not the best time to go about changing the API design.

That being said. I agree that `InstancedConfiguration.defaults()` would be a good to have, but given the number of people that directly use the java API being so low I don't feel strongly either way. One additional note is that in order to synchronously persist a file using the Java API, a user has to user `FileSystemUtils` too. So by that argument shouldn't we also incorporate those utils into the client somehow?

For both points it would require some amount of change to the way we instantiate, populate, and expose the configuration parameters that get read from `alluxio-site.properties`. Based on some initial IntelliJ data there are at least a few hundred call sites which would need to change or be updated.

I think any change at this point to public-facing APIs should require some more time and careful consideration around naming, and which public-facing methods are available rather than being pushed quickly right before release. I think both concerns here are valid but I feel the amount of work to refactor this is larger compared to the somewhat minor usability benefits.

Reply:
Once the API is out, it will be sticky. Meanwhile I could think of some solution in the middle that we could still provide a better public API without really changing the implementation substantially. Let's talk offline

Reply:
Changed to use a slightly nicer API and added warnings to give us some flexibility in the next releases.